frontend/dockerfile/shell/lex.go

```diff
@@ -348,6 +368,26 @@ func (sw *shellWord) processDollar() (string, error) {
 
 			return newValue, nil
 
+		case '?':
+			if !found {
+				if sw.skipUnsetEnv {
+					return fmt.Sprintf("${%s:%s%s}", name, string(modifier), word), nil
+				}
+				var extra string
+				if word != "" {
+					extra = fmt.Sprintf(": %s", word)
+				}
+				return "", errors.Errorf("undefined mandatory variable %s%s", name, extra)
+			}
+			if newValue == "" {
+				var extra string
+				if word != "" {
+					extra = fmt.Sprintf(": %s", word)
+				}
+				return "", errors.Errorf("empty mandatory variable %s%s", name, extra)
```

Comment:
If `word` is defined, it should take the place of the default error message; 

> ${parameter:?[word]}
Indicate Error if Null or Unset. If parameter is unset or null, _**the expansion of word** (or a message indicating it is unset if word is omitted) shall be written to standard error_ and the shell exits with a non-zero exit status. Otherwise, the value of parameter shall be substituted. An interactive shell need not exit.

So if no `word` is set, a default message should be printed:

```bash
echo ${SOMEVAR:?}
SOMEVAR: has not been set
```

But otherwise, `word` should be used;

```
echo ${SOMEVAR:?please set the variable}
SOMEVAR: please set the variable

export MSG=blablabla
echo ${SOMEVAR:?please set the variable ${MSG}}
SOMEVAR: please set the variable blablabla
```

Reply:
@thaJeztah  is the default message implemented yet? The feature is cool but I get:
```docker
ERROR: Missing mandatory value for "container_name" option in service "app":
```
and I think is painful to put an explicit message for every env needed :)